### DISCUSSION TITLE
Can we get Figma files for Tailwind UI components? 🙏

### DISCUSSION THREAD

**Discussion by entity** [tailwindlabs/tailwindcss#2231] [Ideas]
I know this is under both Steve and Adams radar but a figma file for utilising the components would be absolutely amazing.

Times like now I want to be able to play around with components together before copy pasting them in.

I know copying and pasting doesn’t take long, but it’s easier to see colours and such in a playground environment.

Thanks.

**Comment by dillingham** [comment]
My two cents: I think if making design files for each component increases the complexity of releasing components and decreases the turn around time for new components etc, maybe it shouldn't become part of the scope. I originally wanted the file too. But the more I think the more I feel like its an impulse but not actually useful. I can't see making a bunch of changes in a design software and then porting it over any more simply than editing the copied markup and making screenshots. No offense intended to the designers out there wanting a product to make mock ups :)

Almost wonder if a drag and drop component visualizer would satisfy the request..
Would also make for a nice markup jump start while not increasing per component scope.

**Comment by iotekas** [comment]
I always prototype, and do most «customer interactions» with sketch. Before i start coding.

I agree with you on existing codebase. It is easier to implement the component directly in the code.

But before i write any code i always use sketch. 

It would be a huge timesaver with sketch/figma components.

Maybe as a first step only inputs, buttons, and so on.

**Comment by dillingham** [comment]
I guess Im just curious what the extra step's goal is that the browser doesn't satisfy?
Seems like its more of a client presentation request than a dev requirement?

Im starting from scratch and feel its easy personally, but that's subjective both ways.
Maybe you could still screenshot and map screens to simulate interactions?
its a timesaver _if_ you **assemble and tweak** in a software first..
For me, the code is  already the prototype.. and I make changes in browser.

I never noticed people doing this with Bootstrap 🤔 interesting though

**Comment by iotekas** [comment]
I have done both ways, but i find it more effective to use sketch first, and no code.

But it might be about preferences.

**Comment by entity** [comment]
> I have done both ways, but i find it more effective to use sketch first, and no code.
> 
> But it might be about preferences.

I just like to be able to quickly move stuff without need to cut paste code

**Comment by dillingham** [comment]
So whats the next step after you've moved stuff around?
Seems like you essentially have to re-write the code after mocking.
Like it wouldn't directly map to the components anymore..
So it actually adds way more complexity with a lot of changes.

**Comment by entity** [comment]
> So whats the next step after you've moved stuff around?
> Seems like you essentially have to re-write the code after mocking.
> Like it wouldn't directly map to the components anymore..
> So it actually adds way more complexity with a lot of changes.

Personally I like to quickly design something and move stuff around, then TUI would be even easier to do

**Comment by iotekas** [comment]
I see your points, on website design i do it as you describe, i just code.  On application design i do it differently, on the first prototypes, i want the prototypes to be as accurate as possible, but i want to be able to move elements around.

Often this is done with the client, when trying to solve specific challenges

**Comment by joaokamun** [comment]
Also the components need to be designed by Steve and coded by Adam. Figma files stop on the first step (I think), could even be released earlier than the components. In case I need some component not released yet I can write something similar to what will be released and my UI don't become so different from the future released component. Plus, if I want to make a specific component for my project, using Figma with a lot of components is easier to extract small details to design something similar.

**Comment by profsmallpine** [comment]
@adamwathan We've chosen tailwind UI, and are on-boarding a designer. Do you know of any workflow options to create tailwind ui in figma as our designer has no coding experience but wants to use the building blocks to scaffold out UI?

We've tried using: https://www.figma.com/community/plugin/738806869514947558/Tailwind-CSS, but it would still be quite a bit of work to get the components built out to then start designing features.

**Comment by jikkujose** [comment]
Was expecting to find a Figma file when we bought the pack. I think, it would be great if we can have it.

**Comment by simonswiss** [comment]
Figma files are definitely coming, they're being put together as we speak! 👀

**Reply by fitzzz** [reply]
sooooo... can we see them? :)

**Comment by entity** [comment]
> 
> 
> Figma files are definitely coming, they're being put together as we speak! 👀

omgg yes, will these be paid additions or free?

**Comment by pA1nD** [comment]
> Figma files are definitely coming, they're being put together as we speak! 👀

You have no idea how happy that makes me!
It will save me nights of extra work in which I rebuild elements in figma.

If you're putting a beta out first - I'm highly interested in giving feedback!

**Comment by simonswiss** [comment]
@Chazza Figma files are likely to be free for existing Tailwind UI customers 👍

**Comment by pA1nD** [comment]
@simonswiss why is it closed?
Is it done? 😍
Or do you keep track of it somewhere else?

**Comment by michaellee1** [comment]
Hi Simon!  I was wondering if you had another update on this.  This is really important before I buy!

**Reply by entity** [reply]
> Hi Simon! I was wondering if you had another update on this. This is really important before I buy!

Hey, these are currently a thing, just in early stages so not publicly announced as such. If you’re in the TailwindUI discord, they have a channel for such.

If you aren’t, https://tailwindui.com/discord to join.

**Comment by quantuminformation** [comment]
Is there official support for figma - tailwindcss/tailwindUI?
